src/components/ReportInfoView.js

```diff
@@ -79,13 +83,19 @@ class ReportInfoView extends React.Component {
     }
 
     this.setState({
-      didClickNav: true,
       sectionRequiredErrors: requiredErrors
     });
+  }
 
-    if (requiredErrors.length < 1 && this.state.sectionFormatErrors.length < 1) {
-      this.props.handlePageChange(path);
-    }
+  handlePageChange = (path) => {
+    this.setDidClickNav();
+
+    Promise.resolve(this.checkRequiredValidation())
```

Comment:
`checkRequiredValidation()` is synchronous and doesn't return a Promise, so why are you using a Promise here?